parquet/src/column/writer/mod.rs

```diff
@@ -1152,6 +1181,56 @@ fn compare_greater_byte_array_decimals(a: &[u8], b: &[u8]) -> bool {
     (a[1..]) > (b[1..])
 }
 
+/// Truncate a UTF8 slice to the longest prefix that is still a valid UTF8 string, while being less than `max_len` bytes.
+fn truncate_utf8(data: &str, max_len: usize) -> Vec<u8> {
```

Comment:
```suggestion
fn truncate_utf8(data: &str, max_len: usize) -> &str {
```
Might allow preserving the UTF-8ness